api/v1_projects.inc

```diff
@@ -498,6 +498,33 @@ function render_project_page_json($row)
     ];
 }
 
+//---------------------------------------------------------------------------
+// projects/:projectid/transitions
+
+function api_v1_project_transitions($method, $data, $query_params)
+{
+    $sql = sprintf(
+        "
+        SELECT *
+        FROM project_events
+        WHERE projectid = '%s'
+        AND event_type in ('creation', 'transition')
```

Comment:
After looking at all of the possible `event_type`s* I think we should include `deletion` and `archive` as they seem like interesting "transitions" unlike the others which seem more like just "events".

`*` full set from PROD:
```
mysql> select count(*), event_type from project_events group by event_type;
+----------+----------------+
| count(*) | event_type     |
+----------+----------------+
|    67611 | creation       |
|  1020315 | transition     |
|   218565 | edit           |
|     1068 | deletion       |
|    38120 | archive        |
|    24320 | smooth-reading |
|    13872 | add_holds      |
|    13925 | remove_holds   |
+----------+----------------+
8 rows in set (4.04 sec)
```

Reply:
I agree about adding deletion and archive events. This is nit intended to be used for the project history,?

Reply:
> This is not intended to be used for the project history?

No, not really. By focusing on (mostly) textual-free transitions we avoid the issue where `edit` events are both numerous and include what changed in the details which is further unstructured and involves a lot of [localization gymnastics](https://github.com/DistributedProofreaders/dproofreaders/blob/master/project.php#L1207).

Reply:
> After looking at all of the possible `event_type`s* I think we should include `deletion` and `archive` as they seem like interesting "transitions" unlike the others which seem more like just "events".
> 
> `*` full set from PROD:
> 
> ```
> mysql> select count(*), event_type from project_events group by event_type;
> +----------+----------------+
> | count(*) | event_type     |
> +----------+----------------+
> |    67611 | creation       |
> |  1020315 | transition     |
> |   218565 | edit           |
> |     1068 | deletion       |
> |    38120 | archive        |
> |    24320 | smooth-reading |
> |    13872 | add_holds      |
> |    13925 | remove_holds   |
> +----------+----------------+
> 8 rows in set (4.04 sec)
> ```

Footnote: I got curious, because mostly we see a `transition` to `project_delete`. In fact, it looks like `deletion` as an `event_type` was restricted to new projects that had not moved out of the `project_new` state until early 2017. And all deletions (including from `project_new`) since then are logged as transitions to `project_delete`.

Reply:
In fact, digging further, I pulled a handful of projectids on TEST that had an event_type of `deletion`, and all threw a nonexistent project exception (which would have been expected then because the projectids were not kept), and both via `curl` and Swagger, I get

```
{
    "error": "There is no project with projectid 'projectID44dcf4500f8fe'",
    "code": 101
}
```

Is that the expected response?

Reply:
It is, because that project doesn't exist in the `projects` table:
```
mysql> select * from projects where projectid='projectID44dcf4500f8fe';
Empty set (0.00 sec)
```

Reply:
OK, but in that case does it mean we'll never be able to see a deleted project_event, because any projects that were deleted no longer have a row in `projects` and thus won't pass the validator?

Reply:
That's what I would expect. It means that you can't really tell the difference between a real nonexistent projectid (i.e. one that never existed) and one that once existed and was then deleted.

Reply:
> OK, but in that case does it mean we'll never be able to see a deleted project_event, because any projects that were deleted no longer have a row in `projects` and thus won't pass the validator?

That's correct. Ideally a delete would have _actually_ deleted all the projectid records from all tables but we don't have the foreign keys set up (MyISAM didn't support them at the time they were created) and it couldn't delete the projectid-named table either.

Reply:
So... if `deletion` can never be seen, I guess I should take it out of the SQL query and the openapi YAML?

Reply:
Yeah, good call.

Reply:
OK. Done.